File: src/main/java/com/github/solairerove/woodstock/service/TicketService.java
package com.github.solairerove.woodstock.service;

import com.github.solairerove.woodstock.domain.Ticket;
import com.github.solairerove.woodstock.dto.TicketDTO;
import com.github.solairerove.woodstock.service.common.GenericService;

public interface TicketService extends GenericService<Ticket> {

    Ticket create(Long taskId, TicketDTO ticketDTO);

    Ticket update(Long taskId, Long ticketId, TicketDTO ticketDTO);

    Ticket delete(Long taskId, Long ticketId);
}

File: src/main/resources/static/app/service/ticket.service.js
'use strict';

woodstock.factory('TicketService', ['$resource', function ($resource) {
    return $resource('/api/tasks/:taskId/tickets/:ticketId', {}, {
        create: {
            method: 'POST'
        },
        update: {
            method: 'PUT'
        }
    });
}]);
